swift-sdk/IterableActionRunner.swift

```diff
@@ -44,35 +44,47 @@ class IterableActionRunner : NSObject, ActionRunnerProtocol {
         self.urlOpener = urlOpener
     }
     
-    func execute(action: IterableAction) {
+    func execute(action: IterableAction, from source: IterableActionSource) -> Bool {
+        let context = IterableActionContext(action: action, source: source)
+        
         if action.isOfType(IterableAction.actionTypeOpenUrl) {
-            if let data = action.data {
-                if let url = URL(string: data) {
-                    open(url: url, action: action)
-                } else {
-                    ITBError("Could not create url from data: \(data)")
-                }
+            if let urlString = action.data, let url = URL(string: urlString) {
+                return open(url: url, inContext: context)
             } else {
-                ITBError("data is required for action type 'openUrl'")
+                ITBError("Could not create url from action: \(action)")
+                return false
             }
         } else {
-            callCustomActionIfSpecified(action: action)
+            return callCustomActionIfSpecified(action: action, inContext: context)
         }
     }
     
-    private func open(url: URL, action: IterableAction) {
-        if urlDelegate?.handleIterableURL(url, fromAction: action) == true {
-            return
+    private func open(url: URL, inContext context: IterableActionContext) -> Bool {
```

Comment:
Is there anything else that we need to consider in order to handle in-app notification link clicks?

Reply:
I have not thought through in-app scenario here. If we need anything more for in-app we can add this to the IterableActionContext object. The point of having a context object there was so that this is extensible and we can add more "context" as we need.